<a href="https://colab.research.google.com/github/praffs18/EDA-on-All-Indian-movies-IMDb-data/blob/main/EDA_on_All_Indian_movies_IMDb_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

EDA on All Indian movies IMDb Data

This dataset of 15506 Indian movies taken from IMDb. This is all the Indian movies on IMDb as of 16/06/2021.

In [76]:
# import all libraries
import pandas as pd 
import numpy as np

# import visualisation libraries
import matplotlib.pyplot as plt
import seaborn as sns

In [77]:
# Mount the google drive
from google.colab import drive
drive.mount('/content/drive',force_remount = True)

Mounted at /content/drive


In [78]:
# Load the data
path  = "/content/drive/MyDrive/Data Science/EDA Projects/EDA on All Indian movies IMDb Data/IMDb Movies India.csv"
data = pd.read_csv(path,encoding='latin-1')
data.head()

Name    Year Duration            Genre  \
0                                         NaN      NaN            Drama   
1  #Gadhvi (He thought he was Gandhi)  (2019)  109 min            Drama   
2                         #Homecoming  (2021)   90 min   Drama, Musical   
3                             #Yaaram  (2019)  110 min  Comedy, Romance   
4                   ...And Once Again  (2010)  105 min            Drama   

   Rating Votes            Director       Actor 1             Actor 2  \
0     NaN   NaN       J.S. Randhawa      Manmauji              Birbal   
1     7.0     8       Gaurav Bakshi  Rasika Dugal      Vivek Ghamande   
2     NaN   NaN  Soumyajit Majumdar  Sayani Gupta   Plabita Borthakur   
3     4.4    35          Ovais Khan       Prateik          Ishita Raj   
4     NaN   NaN        Amol Palekar  Rajat Kapoor  Rituparna Sengupta   

           Actor 3  
0  Rajendra Bhatia  
1    Arvind Jangid  
2       Roy Angana  
3  Siddhant Kapoor  
4      Antara Mali

In [79]:
data.tail()

Name    Year Duration          Genre  Rating Votes  \
15504  Zulm Ko Jala Doonga  (1988)      NaN         Action     4.6    11   
15505                Zulmi  (1999)  129 min  Action, Drama     4.5   655   
15506            Zulmi Raj  (2005)      NaN         Action     NaN   NaN   
15507        Zulmi Shikari  (1988)      NaN         Action     NaN   NaN   
15508         Zulm-O-Sitam  (1998)  130 min  Action, Drama     6.2    20   

            Director           Actor 1         Actor 2        Actor 3  
15504  Mahendra Shah  Naseeruddin Shah   Sumeet Saigal  Suparna Anand  
15505     Kuku Kohli      Akshay Kumar  Twinkle Khanna    Aruna Irani  
15506     Kiran Thej   Sangeeta Tiwari             NaN            NaN  
15507            NaN               NaN             NaN            NaN  
15508   K.C. Bokadia        Dharmendra      Jaya Prada    Arjun Sarja

In [80]:
# Check how many rows and columns in data
print(f'No. of rows are : {data.shape[0]}')
print(f'No. of feature columns are : {data.shape[1]}')

No. of rows are : 15509
No. of feature columns are : 10


In [81]:
# check all information about the data
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15509 entries, 0 to 15508
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Name      15509 non-null  object 
 1   Year      14981 non-null  object 
 2   Duration  7240 non-null   object 
 3   Genre     13632 non-null  object 
 4   Rating    7919 non-null   float64
 5   Votes     7920 non-null   object 
 6   Director  14984 non-null  object 
 7   Actor 1   13892 non-null  object 
 8   Actor 2   13125 non-null  object 
 9   Actor 3   12365 non-null  object 
dtypes: float64(1), object(9)
memory usage: 1.2+ MB


In [82]:
# Make copy of the data and start doing changes
data_1 = data.copy()
data_1.head()

Name    Year Duration            Genre  \
0                                         NaN      NaN            Drama   
1  #Gadhvi (He thought he was Gandhi)  (2019)  109 min            Drama   
2                         #Homecoming  (2021)   90 min   Drama, Musical   
3                             #Yaaram  (2019)  110 min  Comedy, Romance   
4                   ...And Once Again  (2010)  105 min            Drama   

   Rating Votes            Director       Actor 1             Actor 2  \
0     NaN   NaN       J.S. Randhawa      Manmauji              Birbal   
1     7.0     8       Gaurav Bakshi  Rasika Dugal      Vivek Ghamande   
2     NaN   NaN  Soumyajit Majumdar  Sayani Gupta   Plabita Borthakur   
3     4.4    35          Ovais Khan       Prateik          Ishita Raj   
4     NaN   NaN        Amol Palekar  Rajat Kapoor  Rituparna Sengupta   

           Actor 3  
0  Rajendra Bhatia  
1    Arvind Jangid  
2       Roy Angana  
3  Siddhant Kapoor  
4      Antara Mali

In [83]:
# Remove # sign in the name column
data_1['Name'] = data_1["Name"].str.replace('#','')
data_1.head()

Name    Year Duration            Genre  \
0                                        NaN      NaN            Drama   
1  Gadhvi (He thought he was Gandhi)  (2019)  109 min            Drama   
2                         Homecoming  (2021)   90 min   Drama, Musical   
3                             Yaaram  (2019)  110 min  Comedy, Romance   
4                  ...And Once Again  (2010)  105 min            Drama   

   Rating Votes            Director       Actor 1             Actor 2  \
0     NaN   NaN       J.S. Randhawa      Manmauji              Birbal   
1     7.0     8       Gaurav Bakshi  Rasika Dugal      Vivek Ghamande   
2     NaN   NaN  Soumyajit Majumdar  Sayani Gupta   Plabita Borthakur   
3     4.4    35          Ovais Khan       Prateik          Ishita Raj   
4     NaN   NaN        Amol Palekar  Rajat Kapoor  Rituparna Sengupta   

           Actor 3  
0  Rajendra Bhatia  
1    Arvind Jangid  
2       Roy Angana  
3  Siddhant Kapoor  
4      Antara Mali

In [84]:
# Remove parenthesis in year column
data_1['Year'] = data_1['Year'].str.replace('(','').str.replace(')','')
data_1.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Name  Year Duration            Genre  Rating  \
0                                      NaN      NaN            Drama     NaN   
1  Gadhvi (He thought he was Gandhi)  2019  109 min            Drama     7.0   
2                         Homecoming  2021   90 min   Drama, Musical     NaN   
3                             Yaaram  2019  110 min  Comedy, Romance     4.4   
4                  ...And Once Again  2010  105 min            Drama     NaN   

  Votes            Director       Actor 1             Actor 2          Actor 3  
0   NaN       J.S. Randhawa      Manmauji              Birbal  Rajendra Bhatia  
1     8       Gaurav Bakshi  Rasika Dugal      Vivek Ghamande    Arvind Jangid  
2   NaN  Soumyajit Majumdar  Sayani Gupta   Plabita Borthakur       Roy Angana  
3    35          Ovais Khan       Prateik          Ishita Raj  Siddhant Kapoor  
4   NaN        Amol Palekar  Rajat Kapoor  Rituparna Sengupta      Antara Mali

In [85]:
# Lets check are there any duplicates
data_1.duplicated(keep = False).value_counts()

False    15497
True        12
dtype: int64

This shows that there are few duplicates present in the data

In [86]:
# Check  what are those values are 
duplicate_data = data_1[data_1.duplicated(keep = False)]
duplicate_data

Name  Year Duration     Genre  Rating Votes  \
1249   Arab Ka Sona - Abu Kaalia  1979      NaN    Action     NaN   NaN   
1250   Arab Ka Sona - Abu Kaalia  1979      NaN    Action     NaN   NaN   
1768                     Balidan  1992      NaN     Drama     NaN   NaN   
1769                     Balidan  1992      NaN     Drama     NaN   NaN   
4722     First Time - Pehli Baar  2009      NaN       NaN     NaN   NaN   
4723     First Time - Pehli Baar  2009      NaN       NaN     NaN   NaN   
9712                     Musafir   NaN      NaN  Thriller     NaN   NaN   
9713                     Musafir   NaN      NaN  Thriller     NaN   NaN   
13068                    Shivani  2019      NaN     Crime     NaN   NaN   
13069                    Shivani  2019      NaN     Crime     NaN   NaN   
13307          Slumdog Karodpati  2019  118 min  Thriller     NaN   NaN   
13308          Slumdog Karodpati  2019  118 min  Thriller     NaN   NaN   

                  Director      Actor 1       Actor 2     Actor 3  
1249        Master Bhagwan    Meena Rai    Dara Singh         NaN  
1250        Master Bhagwan    Meena Rai    Dara Singh         NaN  
1768                   NaN          NaN           NaN         NaN  
1769                   NaN          NaN           NaN         NaN  
4722          Raja Bundela  Zeenat Aman   Nitin Arora  Raj Babbar  
4723          Raja Bundela  Zeenat Aman   Nitin Arora  Raj Babbar  
9712           Shiva Dagar          NaN           NaN         NaN  
9713           Shiva Dagar          NaN           NaN         NaN  
13068  Ugresh Prasad Ujala      Santosh           NaN         NaN  
13069  Ugresh Prasad Ujala      Santosh           NaN         NaN  
13307        Rajesh Patole  Udhav Garje  Rahul Gavane   Govindrao  
13308        Rajesh Patole  Udhav Garje  Rahul Gavane   Govindrao

In [87]:
# Remove Duplicate rows
data_2 = data_1.drop_duplicates(keep= 'first')
data_2.shape

(15503, 10)

In [88]:
# Lets check any duplicate in movie name column
data_2.duplicated(subset = 'Name').value_counts()

False    13837
True      1666
dtype: int64

From this we can say that there are duplicate movie names which need to drop

In [89]:
# Remove duplicate rows 
data_2 = data_1.drop_duplicates(subset = 'Name',keep='first')
data_2.shape

(13837, 10)

In [90]:
# Reset the index
data_2.set_index(np.arange(len(data_2)),inplace=True)
data_2

Name  Year Duration            Genre  \
0                                          NaN      NaN            Drama   
1      Gadhvi (He thought he was Gandhi)  2019  109 min            Drama   
2                             Homecoming  2021   90 min   Drama, Musical   
3                                 Yaaram  2019  110 min  Comedy, Romance   
4                      ...And Once Again  2010  105 min            Drama   
...                                  ...   ...      ...              ...   
13832                Zulm Ko Jala Doonga  1988      NaN           Action   
13833                              Zulmi  1999  129 min    Action, Drama   
13834                          Zulmi Raj  2005      NaN           Action   
13835                      Zulmi Shikari  1988      NaN           Action   
13836                       Zulm-O-Sitam  1998  130 min    Action, Drama   

       Rating Votes            Director           Actor 1             Actor 2  \
0         NaN   NaN       J.S. Randhawa          Manmauji              Birbal   
1         7.0     8       Gaurav Bakshi      Rasika Dugal      Vivek Ghamande   
2         NaN   NaN  Soumyajit Majumdar      Sayani Gupta   Plabita Borthakur   
3         4.4    35          Ovais Khan           Prateik          Ishita Raj   
4         NaN   NaN        Amol Palekar      Rajat Kapoor  Rituparna Sengupta   
...       ...   ...                 ...               ...                 ...   
13832     4.6    11       Mahendra Shah  Naseeruddin Shah       Sumeet Saigal   
13833     4.5   655          Kuku Kohli      Akshay Kumar      Twinkle Khanna   
13834     NaN   NaN          Kiran Thej   Sangeeta Tiwari                 NaN   
13835     NaN   NaN                 NaN               NaN                 NaN   
13836     6.2    20        K.C. Bokadia        Dharmendra          Jaya Prada   

               Actor 3  
0      Rajendra Bhatia  
1        Arvind Jangid  
2           Roy Angana  
3      Siddhant Kapoor  
4          Antara Mali  
...                ...  
13832    Suparna Anand  
13833      Aruna Irani  
13834              NaN  
13835              NaN  
13836      Arjun Sarja  

[13837 rows x 10 columns]

In [92]:
# It seems like there are null values in data and those are NaN values lets check that
missing_data = pd.DataFrame(data_2.isna().sum(),columns = ['Nan_values'])
missing_data['Missing %'] = round(missing_data['Nan_values']*100/len(data_2),2)
missing_data

Nan_values  Missing %
Name               0       0.00
Year             498       3.60
Duration        7239      52.32
Genre           1590      11.49
Rating          6684      48.31
Votes           6683      48.30
Director         465       3.36
Actor 1         1439      10.40
Actor 2         2131      15.40
Actor 3         2804      20.26

From this we can see that there are lot of NaN Values are present in the data
1. Duration, Rating & Votes columns has Nan Values Approximately 52%, 48% & 48% respectively. which are huge in numbers.